### PR TITLE
Part 2: PayPal Checkout Integration

### PR DESCRIPTION
**Pull request by mufudzimasaire**
## Overview

Github Issue: https://github.com/shiftcommerce/shift-front-end-react/issues/845

This PR add PayPal logic for updating the order total to include the selected shipping method.

Dependent on
[Shift Front End React](https://github.com/shiftcommerce/shift-front-end-react/pull/846)

## QA

1. checkout branch `issue/790_paypal_checkout_integration_part_2`

2. In the `shift-front-end-react` project, checkout branch `issue/790_paypal_checkout_integration`. Boot app (assumption is that you are using yalc)

3. Add items to cart and navigate to the first step of checkout - `payment-method` selection

4. Check that the `Continue Shopping` btn in the cart should be full-width

5. Check that the `Payment Method` step is present and highlighted

6. Check that there are payment method options displayed on the page

7. Check that  clicking the `Pay By Credit/Debit Card` button results in the page navigating to checkout step 2 - `Shipping Address`

8. Check that clicking the `PayPal` button in the page navigates to the PayPal website. 

9. Complete all steps on PayPal and click the continue button

10. When you have been redirected back to the site, check that the shipping address summary is present and that you are redirect to checkout 3 - shipping method. Also ensure that the `Edit` address button is not present and that clicking the shipping address step in the checkout header steps, redirects you back to checkout 1 - Payment Method selection

11. Continue to the review page and place order

review page

12. Confirm order

Order confirmation page

## Deployment
The following environment variables are required for this feature to work:
```
PAYPAL_CLIENT_ID='test'
PAYPAL_CLIENT_SECRET='secret'
ENABLE_PAYPAL_LIVE_ENVIRONMENT=false
```

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -28,6 +29,7 @@
     "@babel/plugin-transform-runtime": "^7.2.0",
     "@babel/preset-env": "^7.3.1",
     "@babel/preset-react": "^7.0.0",
+    "@paypal/checkout-server-sdk": "^1.0.0",
```
Comment (mufudzimasaire) [inline package.json]: Added in dev dependencies as its required for tests

File: src/actions/action-types.js [hunk 2]
```diff
@@ -36,10 +36,12 @@ export const SHIPPING_ADDRESS_CREATED = 'SHIPPING_ADDRESS_CREATED'
...
 export const SET_CHECKOUT_INPUT_SHOWN = 'SET_CHECKOUT_INPUT_SHOWN'
-export const SET_FILTER_CATEGORY = 'SET_FILTER_CATEGORY'
 export const SET_CHECKOUT_BILLING_ADDRESS = 'SET_CHECKOUT_BILLING_ADDRESS'
 export const SET_CHECKOUT_SHIPPING_ADDRESS = 'SET_CHECKOUT_SHIPPING_ADDRESS'
+export const SET_FILTER_CATEGORY = 'SET_FILTER_CATEGORY'
 export const SET_MENU = 'SET_MENU'
 export const SET_PAYMENT_ERROR = 'SET_PAYMENT_ERROR'
-export const SET_PAYMENT_METHOD = 'SET_PAYMENT_METHOD'
-export const SET_PAYPAL_ORDER_DETAILS = 'SET_PAYPAL_ORDER_DETAILS'
+export const PATCH_PAYPAL_ORDER = 'PATCH_PAYPAL_ORDER'
```
Comment (owensmooth) [inline src/actions/action-types.js]: can keep this file in alphabetical order

File: src/actions/order-actions.js [hunk 3]
```diff
@@ -79,7 +79,8 @@ export function convertCheckoutToOrder (cart, paymentMethod, order) {
   return {
     data: orderPayload,
     payment_method: paymentMethod,
-    card_token: order.cardToken
+    card_token: order.cardToken,
+    paymentAuthorizationID: order.paymentAuthorization.id
```
Comment (Azfletch) [inline src/actions/order-actions.js]: all other keys here seem to be snake_case. could this be changed to payment_authorization_id?

File: src/pages/checkout/payment-method.js [hunk 4]
```diff
@@ -101,7 +101,8 @@ export class PaymentMethodPage extends Component {
   handlePayPalOrderResponse (order) {
     const payer = order.payer
```
Comment (Azfletch) [inline src/pages/checkout/payment-method.js]: Could this line use destructuring?

```suggestion
    const { payer } = order
```

File: src/pages/checkout/shipping-method.js [hunk 5]
```diff
@@ -84,9 +120,10 @@ export class ShippingMethodPage extends Component {
     this.setState({ selectedShippingMethod: shippingMethod })
   }
 
-  nextSection (eventType) {
-    const { checkout, thirdPartyPaymentMethods} = this.props
-    if (thirdPartyPaymentMethods.includes(checkout.paymentMethod)) {
+  nextSection () {
+    const { setCurrentStep, thirdPartyPaymentMethods} = this.props
```
Comment (Azfletch) [inline src/pages/checkout/shipping-method.js]: needs a space after `thirdPartyPaymentMethods`

File: src/pages/checkout/shipping-method.js [hunk 6]
```diff
@@ -96,11 +133,14 @@ export class ShippingMethodPage extends Component {
   }
 
   continueButtonProps () {
+    const { thirdPartyPaymentMethods} = this.props
```
Comment (Azfletch) [inline src/pages/checkout/shipping-method.js]: same as above

File: src/pages/checkout/shipping-method.js [hunk 7]
```diff
@@ -110,7 +150,7 @@ export class ShippingMethodPage extends Component {
 
   render () {
     const { cart } = this.props
-    const { checkout: { shippingAddress } } = this.props
+    const { thirdPartyPaymentMethods } = this.props
```
Comment (Azfletch) [inline src/pages/checkout/shipping-method.js]: can these two can be on the same line?

```suggestion
    const { cart, thirdPartyPaymentMethods } = this.props
```

File: src/pages/order.js [hunk 8]
```diff
@@ -54,6 +69,31 @@ class OrderPage extends Component {
     )
   }
 
+  renderPaymentMethodInformation (order) {
+    if (this.state.paymentMethod === "PayPal") {
```
Comment (Azfletch) [inline src/pages/order.js]: lets try and stick to single quotes throughout where we can

**Pull request merged**